ingredient/ingredient-postgresql/src/plugin.ts

```diff
@@ -1,28 +1,145 @@
+                `/providers/Microsoft.Network/privateDnsZones/${privateDnsZoneName}`};
+        }
+        
+        let vnetData: VnetData = {
+            value: {
+                virtualNetworkName: params.virtualNetworkName.value,
+                virtualNetworkId: vNet.id!,
+                subnetName: params.subnetName.value,
+                virtualNetworkAddressPrefix: subnetPropertiesGet.addressPrefix!,
+                virtualNetworkResourceGroupName: params.virtualNetworkResourceGroup.value,
+                location: await util.current_region(), // maybe primary_region()?
+                subscriptionId: this._ctx.Environment.authentication.subscriptionId,
+                subnetProperties: subnetPropertiesGet,
+                subnetNeedsUpdate: false,
+                isNewVnet: false,
+                usePrivateDnsZone: (this._access === "private"),
+                isNewPrivateDnsZone: dnsZoneIsNew, 
+                privateDnsResourceGroup: params.virtualNetworkResourceGroup.value,
+                privateDnsSubscriptionId: this._ctx.Environment.authentication.subscriptionId,
+                privateDnsZoneName: privateDnsZoneName,
+                linkVirtualNetwork: true,
+                Network: {
+                    DelegatedSubnetResourceId: subnetPropertiesGet.id!,
+                    PrivateDnsZoneArmResourceId: dnsZone.id 
+                }
+            }
+        };
+
+        return vnetData;
+    }
+
+    private validateBakeParams(params: any) {
+        const validAccesses = ["public", "private"];
+        if (!validAccesses.includes(this._access)) throw new Error("Parameter 'access' must be set to \"public\" or \"private\".");
+        
+        // This gets checked by the regular ARM validation anyway but might as well catch it early here.
+        if (!params.serverName || !params.administratorLogin || !params.administratorLoginPassword) {
+            throw new Error("serverName, administratorLogin, and administratorLoginPassword must be defined in the Bake parameters.");
+        }
+
+        // private access requires some special data for subnet
+        if (this._access == "private") {
+            if (!params.subnetName || !params.virtualNetworkName || !params.virtualNetworkResourceGroup) {
+                throw new Error("subnetName, virtualNetworkName, and virtualNetworkResourceGroup must be defined in the Bake Parameters for 'private' access");
+            } 
+        }
+    }
+
+    private trimParametersForARM(params: any) {
+        for (var param in params) {
+            if (!this._armTemplate.parameters.hasOwnProperty(param)) {
+                this._logger.log(`Removing unneeded parameter ${param}`) // todo rethink this. we want users to know when they added something that isn't being used, but we don't necessarily want to log when a parameter like virtualNetworkResourceGroup is removed.
```

Comment:
Same for this todo

Reply:
I think I'll just remove the log call. I don't think it is a big deal if somebody accidentally adds a parameter that isn't used.